Added data for TS0505B_1

malorezzanzo64 opened the pull request:
I have some doubts about the manufacturer:
- I bought it on [aliexpress](https://it.aliexpress.com/item/1005004937378562.html) where the manufacturer seems to be KonouYouKo, while on Zigbee2MQTT the manufacturer is TuYa and the zigbee manufacturer is _TZ3210_r5afgmkl.
In model.json I wrote Tuya *TS0505B_1* as the name and *_TZ3210_r5afgmkl*, *TS0505B_1*, *konouyouko* as aliases

bramstroker commented:
What a mess with all this Tuya stuff, and chinese manufacturers.
KonouYouKo doesn't seem to be a brand or manufacturer, but just an aliexpress store. Google it, doesn't give any results.
TS0505B_1 seems also quite generic.

Powercalc looks at the device information of the light for auto discovery.
So please have a look first in device page what's reported there.
Then we can decide what to do.

<img width="377" alt="Screenshot 2023-09-03 at 13 16 56" src="https://github.com/bramstroker/homeassistant-powercalc/assets/2345875/003b0e65-e77a-4cdd-91fe-e1a57dc33506">

`name` field is only used in the supported model listing, so it's best practice to just set that to full marketed name.

For the model matching first the directory name is checked, when that doesn't matches powercalc tries any of the defined aliases.

These aliases are not used for the manufacturers, so `KonouYouKo` must be removed there.

bramstroker commented:
I think it's because your specific model is not known by library Zigbee2Mqtt uses.
I don't use Zigbee and this integration myself but I assume that is managed here:
https://github.com/Koenkk/zigbee-herdsman-converters
Should probably be added something here: https://github.com/Koenkk/zigbee-herdsman-converters/tree/master/src/devices
You could have a look into doing that, or create issue or ask for help there.

It will probably fallback to some kind of generic zigbee light (TS0505B_1).

So currently we cannot use auto discovery as TS0505B_1 is way to generic and could be hundreds of different lights.

We can still adopt the light model into the database, and you can just setup manually using this procedure:
https://homeassistant-powercalc.readthedocs.io/en/latest/sensor-types/virtual-power-library.html

To get the PR merged following needs to be taken care of:
- Remove all aliases from `model.json` as they all are not usable for autodiscovery.
- You'll need to look for actual model number (maybe there is something on the box, or maybe official Tuya app indicates a model correctly). When you have found one rename the folder from `TS0505B_1` to this model number
- Set the name to something the user can recognize. For example `Konouyouko Smart Light E27 LED Lights Bulb RGB+WW+CW`

malorezzanzo64 commented:
I bought another zigbee coordinator to use with ZHA and added the light bulb and this is the device info with ZHA:
![zigbee info](https://github.com/bramstroker/homeassistant-powercalc/assets/96385154/7c91deb7-4487-4474-8394-557bbf75f69d)
It is the same as that of Z2M. I asked the aliexpress store info about the maker and this is their answer:
![konouyouko response](https://github.com/bramstroker/homeassistant-powercalc/assets/96385154/4848b921-2c08-44ca-baa4-2d8e65432653)

I searched for the zigbee model in https://github.com/Koenkk/zigbee-herdsman-converters and the only thing I found is this [issue](https://github.com/Koenkk/zigbee2mqtt/issues/13770)

I updated the name and deleted the aliases as you said

bramstroker commented:
Thanks for your efforts.
Unfortunately we still have an issue with `TS0505B` being to generic. There could potentially be multiple different bulbs identified by Zigbee2Mqtt as `TS0505B`. This would cause the profile you submitted to be loaded for all lights which identify themself as `TS0505B`. 

I think I might need to reject the PR.
But you can still use your measurements in your own installation.
Just put the files in `config/powercalc-custom-models` directory.
Also see https://homeassistant-powercalc.readthedocs.io/en/latest/library/protocol.html.

bramstroker commented:
Unfortunately need to reject this PR for same reasons as mentioned in #1928.
You can use the measurements in your own installation as mentioned above.
Thanks for your efforts.
